File: frontend/src/models/RootDto.ts
import type { CommitDto } from "@/models/CommitDto"
import type { SymbolDto } from "@/models/SymbolDto"
import type { IndexRootDto } from "@/models/IndexRootDto"
import type { MetaDto } from "@/models/MetaDto"

export interface RootDto {
  meta: MetaDto;
  commits: CommitDto[];
  symbols: SymbolDto[];
  indices: IndexRootDto;
}

File: backend/analyzer/src/main/java/com/mategka/dava/analyzer/serialization/model/RootDto.java
package com.mategka.dava.analyzer.serialization.model;

import lombok.Builder;
import lombok.NonNull;
import lombok.Value;
import org.jetbrains.annotations.NotNull;

import java.util.List;

@Value
@Builder
public class RootDto {

  @NonNull
  List<@NotNull CommitDto> commits;

  @NonNull
  List<@NotNull SymbolDto> symbols;

  //@NonNull
  //IndexRootDto indices;

}
